lib/src/stream_sink_transformer/reject_errors.dart

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+/// A [StreammSink] wrapper that rejects all errors passed into the sink.
+class RejectErrorsSink<T> implements StreamSink<T> {
+  /// The inner sink being wrapped.
+  final StreamSink<T> _inner;
+
+  @override
+  Future<void> get done => _doneCompleter.future;
+  final _doneCompleter = Completer();
```

Comment:
`Completer<void>();`. No reason to not have the correct type, even if it's unlikely that someone will print `sink.done.runtimeType`.
(I prefer writing the type for fields, even if they're not public, but I guess inference can handle this one).